.ci/.azure-pipelines-steps.yml

```diff
@@ -11,7 +11,7 @@ steps:
 - script: npm install
   displayName: npm install
 
-- script: npm run coveralls
+- script: istanbul cover _mocha --report lcovonly && istanbul-coveralls
```

Comment:
I know I'm being a little paranoid here but what would happen if someone changed the `package.json` to point `istanbul` or `istanbul-coveralls` at some git repository?

Reply:
Can someone do that? My understanding of how npm works is that it makes that sort of behavior impossible. They'd have to change the dependency name in `package.json` at which point it would no longer be added to the path as `Istanbul` or `Istanbul-coveralls`, it would get added as something else.

Reply:
I'm allowed to do something like:
```json
{
    "devDependencies": {
        "istanbul": "github:phated/malicious-package"
    }
}
```
And as long as "malicious-package" contained an `istanbul` bin, it'd run.

Reply:
Maybe this step needs to forcefully install the packages from npm.  We can probably achieve that by using `npx --ignore-existing istanbul && npx --ignore-existing istanbul-coveralls` - Thoughts?

Reply:
Ah, I see what you mean now - I guess that could be an issue. I think that solution should work in theory, I'm having a bit of trouble getting it to work though because it looks like npx doesn't work correctly on lower versions of npm or Node - see [build here](https://dev.azure.com/gulpci/gulp/_build/results?buildId=95).

We can keep looking for a solution here, to be honest I'm not sure how necessary this is though. I don't think there's anything a malicious actor could do with this token outside of messing with your code coverage results - the work to do that seems to far outweighs the benefits, especially since you can just regen the token and delete the bad builds. In the codecov docs they don't even recommend keeping it as a secret variable even though this would affect all CI's outside of Travis and I think most repos that use this without Travis just keep it as plaintext.

If you want I can keep trying to tweak npx though, I'm happy to continue to work on this if we think its worthwhile. Just don't want to overcomplicate things if we don't need to.